modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/FillFactorMetricTest.java

```diff
@@ -0,0 +1,184 @@
+package org.apache.ignite.internal.processors.cache.persistence.pagemem;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.ignite.DataRegionMetrics;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.spi.discovery.tcp.TcpDiscoverySpi;
+import org.apache.ignite.spi.discovery.tcp.ipfinder.vm.TcpDiscoveryVmIpFinder;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+/**
+ * Tests for fillFactor metrics.
+ */
+public class FillFactorMetricTest extends GridCommonAbstractTest {
+    /** */
+    private static final TcpDiscoveryVmIpFinder IP_FINDER = new TcpDiscoveryVmIpFinder(true);
+
+    /** */
+    private static final String MY_DATA_REGION = "MyPolicy";
+
+    /** */
+    private static final String MY_CACHE = "mycache";
+
+    /** */
+    public static final int NODES = 2;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setDiscoverySpi(new TcpDiscoverySpi().setIpFinder(IP_FINDER))
+            .setDataStorageConfiguration(
+                new DataStorageConfiguration().setDataRegionConfigurations(
+                    new DataRegionConfiguration()
+                        .setName(MY_DATA_REGION)
+                        .setInitialSize(100 * 1024L * 1024L)
+                        .setMaxSize(200 * 1024L * 1024L)
+                        .setMetricsEnabled(true)
+                ));
+    }
+
+    /** */
+    protected CacheConfiguration<Object, Object> cacheCfg() {
+        return new CacheConfiguration<>()
+            .setName(MY_CACHE)
+            .setDataRegionName(MY_DATA_REGION)
+            .setAffinity(new RendezvousAffinityFunction().setPartitions(16));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected long getTestTimeout() {
+        return 300_000;
+    }
+
+    /**
+     * Records counter.
+     */
+    private volatile int i = 0;
+
+    /**
+     * Last fill factor values.
+     */
+    private final float[] currentFillFactor = new float[NODES];
+
+    /**
+     * throws if failed.
+     */
+    public void testName() throws Exception {
```

Comment:
Could you please optimze test to avoid long running test on TC? Also please rename testName()